src/set.rs

```diff
@@ -600,3 +602,16 @@ where
         }
     }
 }
+
+#[cfg(feature = "serialize")]
+impl<T, S> Serialize for HashSet<T, S>
```

Comment:
So we're also missing `Deserialize` for `HashSet`, right?

Reply:
Yes, we're still missing that.

Reply:
I'd like to get some input on the implementation of `Deserialize` before I proceed. Is the `HashMapVisitor` business the right way of going about this?

Reply:
I really feel like we're abusing the type system in here, since `HashMapVisitor`does no represent a value. This is apparently how the developers of `serde`expect us to do things, though.

Reply:
I don't think this is really abusing the type system -- it is helping it! In general, my inclination here is to pretty much exactly whatever [`hashbrown` does for `serde`](https://github.com/rust-lang/hashbrown/blob/master/src/external_trait_impls/serde.rs).

Reply:
@jonhoo I'm quite pick about this kind of stuff hehe. I'm quite a fan of the phrase _let data be data_ (and it's counterpart _let procedures be procedures_), so I'm not that comfortable with a type whose only purpose is to encapsulate behaviour.

However, I believe we should go with whatever is the norm. There's no point on trying to fight `serde`'s API and Rust's semantics (which I quite like overall) because of a simple issue like this.

Anyway, enough of my random rambling on semantics. I guess this conversation could be resolved? We've already implemented `Deserialize` for `HashSet`.

Reply:
I actually don't agree — I think it is extremely useful to be able to leverage the type system to express constraints that are not data-dependent. Especially if it makes illegal states impossible to express. But yes, resolved!